src/app/directives/datatable/datatable.directive.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  AfterContentInit,
+  Directive,
+  ElementRef,
+  Input,
+  OnInit
+} from "@angular/core";
+import { DatatableComponent } from "@swimlane/ngx-datatable";
+
+@Directive({
+  selector: "[appDatatable]"
+})
+export class DatatableDirective implements OnInit, AfterContentInit {
+  @Input() datatable: DatatableComponent;
+  @Input() defaults: Partial<DatatableComponent>;
+  private datatableConfig: Partial<DatatableComponent>;
+
+  constructor(private datatableRef: ElementRef) {}
+
+  ngOnInit(): void {
+    this.datatableRef.nativeElement.classList.add("bootstrap");
+
+    this.datatableConfig = Object.assign(
+      {
+        footerHeight: 50,
+        headerHeight: 50,
+        limit: 25,
+        rowHeight: "auto",
+        scrollbarH: true,
+        reorderable: false
+      },
+      this.defaults
+    );
+  }
+
+  ngAfterContentInit() {
+    for (const key of Object.keys(this.datatableConfig)) {
+      this.datatable[key] = this.datatableConfig[key];
```

Comment:
This is problematic. In your PR description, you said: "To override defaults". Defaults don't need to be overridden (do they?).

I'd suggest this is implemented like this:

1. get host component
1. `for property in this.datatableConfig`
    1. if the host's element has attribute of the same name, continue
    1. else, set the property on host

Reply:
Ngx-datatable comes with its own predefined defaults. Checking if a key exists before overwriting it would be hit and miss resulting in obscure issues.

Reply:
I think you misunderstand.

Order of priority for inputs are (highest wins):

1. ngx defaults
2. our defaults
3. special-case override of our defaults <-- I don't think we need this
3. a value set by an attribute in the template

The rules are (highest wins):

1. if no default set by us, use theirs
2. if our default set, use that
3. if our default are overridden,  use those (<-- still don't think we need this)
4. if an attribute is set use it

With your current method, you could well see that an attribute is ignored because you always assign all of our defaults to the host component.

Rather, to respect rule 4, you should only assign one of our defaults to the host component IFF an attribute of the same name is not already set in the template.

Reply:
The only issue is that I have no method of determining what attributes are set, I can only see the values assigned to the component after it has been initialized. `<ngx-datatable>` default values, and attributes set by the parent component.

For example, the default headerHeight set by `ngx-datable` is 75. If I execute the following code in the parent component:

```html
<ngx-datatable
      #table
      bawDatatableDefaults
      [headerHeight]=150">
```

The `DatatableDirective` will only see headerHeight=150 when it checks the `ngx-datatable` component. Unless I maintain a list of default values set for the `DatatableComponent`, I have no method of knowing what changed. Furthermore, if the parent component sets a value to a `DatatableComponent` how can I tell that it has been changed?

Reply:
The workaround I can think of is to set `@Input()`'s for all of the default config values the user can change. This would allow me to do what you want, however it would be fairly repetitive.

Eg:

```typescript
export class DatatableDirective implements OnInit, AfterContentInit {
  @Input() footerHeight: number;

  private datatableConfig: Partial<DatatableComponent>;

  constructor(
    @Host() private datatable: DatatableComponent,
    private datatableRef: ElementRef
  ) {}

  ngOnInit(): void {
    this.datatableRef.nativeElement.classList.add("bootstrap");
    this.datatableConfig.footerHeight = this.footerHeight || 50;
  }

  ngAfterContentInit() {
    for (const key of Object.keys(this.datatableConfig)) {
      this.datatable[key] = this.datatableConfig[key];
    }
  }
}

```